### PR TITLE
[analyzer] Unknown GCC flags

### PR DESCRIPTION
**Pull request by bruntib**

### CODE REVIEW

File: analyzer/codechecker_analyzer/buildlog/log_parser.py [hunk 1]
```diff
@@ -245,7 +246,7 @@
     re.compile('(' + '|'.join(INCLUDE_OPTIONS_MERGED) + ')')
 
 
-PRECOMPILATION_OPTION = re.compile('-(E|M[G|T|Q|F|J|P|V|M]*)$')
+PRECOMPILATION_OPTION = re.compile('-(E|M[G|T|Q|F|J|P|V|M|D]*)$')
```
Comment (csordasmarton) [inline analyzer/codechecker_analyzer/buildlog/log_parser.py:249]: The title of the commit says that this patch is about unknown gcc flags. `-D` is not an unknown flag? Please move this change to a different commit.

By the way why don't we list all the pre-processor options here (https://gcc.gnu.org/onlinedocs/gcc/Preprocessor-Options.html) such as `-U`, `-H` etc.?

Comment (bruntib) [reply analyzer/codechecker_analyzer/buildlog/log_parser.py:249]: I updated the commit message's title.

The goal of `PRECOMPILATOPN_OPTION` is to determine whether that build action reaches the compilation phase thus it is a candidate for analysis. If you have compilation action containing `-E` for example then compilation will not happen, but the process stops after precompilation. I'm not sure if these `M*` flags all have the same behavior, sometime we could review it.

Comment (csordasmarton) [reply analyzer/codechecker_analyzer/buildlog/log_parser.py:249]: @bruntib With the `-D` option you predefine a macro for the preprocessor but compilation phase also happens:
`g++ -DVAR=1 -c main.cpp`. So the processor will not stop after precompilation.

Comment (bruntib) [reply analyzer/codechecker_analyzer/buildlog/log_parser.py:249]: That's true, but this regex describes `-MD` flag. This regex with these vertical characters are tricky, because `M` is followed by the opening square bracket `[`, not a pipe `|`.

Comment (bruntib) [reply analyzer/codechecker_analyzer/buildlog/log_parser.py:249]: I mean the first `M` at the beginning of the regex :)

Comment (csordasmarton) [reply analyzer/codechecker_analyzer/buildlog/log_parser.py:249]: Oh, okay, now I understand. It is very tricky :smile:
